diff --git a/radicale/app/propfind.py b/radicale/app/propfind.py
index 0c9a0cf..a529750 100644
--- a/radicale/app/propfind.py
+++ b/radicale/app/propfind.py
@@ -159,6 +159,6 @@ def xml_propfind_response(
             uri = str(share['PathOrToken']) + uri.removeprefix(share['PathMapped'])
         if share_bday_automap and not uri.endswith("/"):
-            uri = uri.rstrip(".vcf") + ".ics"
+            uri = uri.removesuffix(".vcf") + ".ics"
 
     href.text = xmlutils.make_href(base_prefix, uri)
     response.append(href)
diff --git a/radicale/item/__init__.py b/radicale/item/__init__.py
index dfe7054..f74a968 100644
--- a/radicale/item/__init__.py
+++ b/radicale/item/__init__.py
@@ -592,6 +592,6 @@ class Item:
 
         href = self.href
         if href is not None:
-            href = href.rstrip(".vcf") + ".ics"
+            href = href.removesuffix(".vcf") + ".ics"
 
         etag = self.etag
         # replace 14 leading chars of etag "<hexdigits>" by special format bda0YYYYMMDD00
